File: openapi-parser/src/main/java/io/openapiparser/model/v31/Extensions.java
/*
 * Copyright 2021 https://github.com/openapi-processor/openapi-parser
 * PDX-License-Identifier: Apache-2.0
 */

package io.openapiparser.model.v31;

import java.util.Map;

/**
 * the <em>Specification Extensions</em> object.
 *
 * <p>See specification:
 * <a href="https://spec.openapis.org/oas/v3.1.0.html#specification-extensions">
 *   4.9 Specification Extensions
 * </a>
 */
public interface Extensions {

    /**
     * map of all extension properties.
     *
     * @return map of extension properties
     */
    Map<String, Object> getExtensions ();
}

File: openapi-parser/src/test/kotlin/io/openapiparser/model/v3x/Extensions.kt
/*
 * Copyright 2021 https://github.com/openapi-processor/openapi-parser
 * PDX-License-Identifier: Apache-2.0
 */

package io.openapiparser.model.v3x

import io.kotest.core.spec.style.stringSpec
import io.kotest.matchers.maps.shouldBeEmpty
import io.kotest.matchers.nulls.shouldNotBeNull
import io.kotest.matchers.shouldBe

fun <O> testExtensions(
    name: String,
    build: (content: String) -> O,
    extract: (o: O) -> Map<String, Any>
) = stringSpec {

        "gets $name extension values" {
            val source = """
          any: value
          x-foo: "foo extension"
          x-bar: "bar extension"
        """

            val extensions = extract(build(source))
            extensions.shouldNotBeNull()
            extensions.size shouldBe 2
        }

        "gets $name empty extension values" {
            extract(build("{}")).shouldBeEmpty()
        }
    }
